src/client/Client.cc

```diff
@@ -7952,6 +7956,10 @@ int Client::readlinkat(int dirfd, const char *relpath, char *buf, loff_t size, c
     return r;
   }
 
+  if (!strcmp(relpath, "")) {
+    return _readlink(dirinode.get(), buf, size);
```

Comment:
readlink on the containing dir?

Also see: https://tracker.ceph.com/issues/51495

Reply:
> readlink on the containing dir?

Yes, that is the expected behaviour as per the following description from man [readlinkat(2)](https://www.man7.org/linux/man-pages//man2/readlinkat.2.html).

"Since Linux 2.6.39, pathname can be an empty string, in which case the call operates on the symbolic link referred to by dirfd (which should have been obtained using [open(2)](https://www.man7.org/linux/man-pages//man2/open.2.html) with the O_PATH and O_NOFOLLOW flags)."

> Also see: https://tracker.ceph.com/issues/51495

Yeah, I've updated the change to return ENOENT in case dirfd doesn't point to a symbolic link. Let me know if I missed something else.